core/src/main/java/org/apache/iceberg/SchemaParser.java

```diff
@@ -248,8 +296,122 @@ public static Schema fromJson(String json) {
       try {
         return fromJson(JsonUtil.mapper().readValue(jsonKey, JsonNode.class));
       } catch (IOException e) {
-        throw new RuntimeIOException(e);
+        throw new RuntimeException(e);
       }
     });
   }
+
+  private static String toJsonString(Map<String, Object> map) {
```

Comment:
I have left a comment in the testing section to validate if this works for deeply nested types properly.

Reply:
added test and it works